app/controllers/revisions_controller.rb

```diff
@@ -54,28 +62,37 @@ def load_node
     initialize_nodes_sidebar
   end
 
-  def load_record
-    @record = if params[:card_id]
-                @list.cards.find(params[:card_id])
-              elsif params[:evidence_id]
-                @node.evidence.find(params[:evidence_id])
-              elsif params[:issue_id]
-                Issue.find(params[:issue_id])
-              elsif params[:note_id]
-                @node.notes.find(params[:note_id])
-              else
-                raise 'Unable to identify record type'
-              end
-  rescue ActiveRecord::RecordNotFound
-    flash[:error] = 'Record not found'
-    redirect_to :back
+  def set_revision_params
+    @revision_params = { :"#{@record.class.name.downcase}_id" => @record.id }
+    @revision_params[:node_id] = @node.id if @node
+
+    if @list
+      @revision_params[:board_id] = @record.board.id
+      @revision_params[:list_id] = @record.list.id
+    end
   end
 
-  def load_sidebar
-    if params[:evidence_id] || params[:note_id]
-      load_node
-    elsif params[:card_id]
+  def set_variables_from_params
+    if params[:card_id]
       load_list
+      @record = @list.cards.find(params[:card_id])
+    elsif params[:evidence_id]
+      load_node
+      @record = @node.evidence.find(params[:evidence_id])
+    elsif params[:issue_id]
+      load_issues
+      @record = @issue
+    elsif params[:note_id]
+      load_node
+      @record = @node.notes.find(params[:note_id])
+    else
+      raise 'Unable to identify record type'
     end
+
+    set_revision_params
+
+  rescue ActiveRecord::RecordNotFound
```

Comment:
I'm not fond of using errors as control flow and we may be better off using if else here instead. Instead of using `find` we can use `find_by_id` so that it returns nil if the record is missing instead of throwing an error. Then, we can check for the existence of `@record` here instead.

Reply:
I removed this rescue block completely since there is no valid case for a record to not be found.